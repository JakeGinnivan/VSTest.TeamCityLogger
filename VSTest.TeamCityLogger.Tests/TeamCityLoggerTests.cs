using System;
using System.Collections.ObjectModel;
using System.Diagnostics;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestPlatform.ObjectModel;
using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
using NSubstitute;
using Xunit;

namespace VSTest.TeamCityLogger.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class TeamCityLoggerTests
    {
        [Fact]
        public void SimpleScenario()
        {
            var sb = new StringBuilder();
            Trace.Listeners.Clear();
            var logger = new TeamCityLogger();
            Trace.Listeners.Add(new TextWriterTraceListener(new StringWriter(sb)));
            var events = Substitute.For<TestLoggerEvents>();
            logger.Initialize(events, string.Empty);

            events.TestResult +=
                Raise.EventWith(
                    new TestResultEventArgs(
                        new TestResult(new TestCase("Test.FullyQualified.Name", new Uri("logger://TeamCityLogger"),
                            "c:\\TestAssembly1.dll"))));
            events.TestResult +=
                Raise.EventWith(
                    new TestResultEventArgs(
                        new TestResult(new TestCase("Test.FullyQualified.Name2", new Uri("logger://TeamCityLogger"),
                            "c:\\TestAssembly2.dll"))));

            events.TestRunComplete +=
                Raise.EventWith(
                    new TestRunCompleteEventArgs(Substitute.For<ITestRunStatistics>(), false, false, null,
                        new Collection<AttachmentSet>(), TimeSpan.FromSeconds(20)));

            Approvals.Verify(Regex.Replace(Regex.Replace(sb.ToString(), " timestamp='[^']*'", string.Empty), " flowId='[^']*'", string.Empty));
        }
    }
}
