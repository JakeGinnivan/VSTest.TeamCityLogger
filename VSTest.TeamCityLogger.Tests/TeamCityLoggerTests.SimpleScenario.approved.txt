##teamcity[testSuiteStarted name='VSTest']
##teamcity[testSuiteStarted name='TestAssembly1.dll']
##teamcity[testStarted name='Test.FullyQualified.Name' captureStandardOutput='false']
##teamcity[testFinished name='Test.FullyQualified.Name' duration='0']
##teamcity[testSuiteFinished name='TestAssembly1.dll']
##teamcity[testSuiteStarted name='TestAssembly2.dll']
##teamcity[testStarted name='Test.FullyQualified.Name2' captureStandardOutput='false']
##teamcity[testFinished name='Test.FullyQualified.Name2' duration='0']
##teamcity[testSuiteFinished name='TestAssembly2.dll']
##teamcity[testSuiteFinished name='VSTest']
Total Executed: 0
Total Passed: 0
Total Failed: 0
Total Skipped: 0
